sig-architecture/api-review-process.md

```diff
@@ -85,94 +61,83 @@ Because expert reviewer bandwidth is extremely limited, the process provides a c
+    * Requests are triaged by API approvers/reviewers/moderators [regularly](#review-lifecycle-timing), and added to the project board if prereqs have been completed
+    * As requests are added to the project board, that is reflected in the sidebar of the issue or PR, along with the current status (backlog, assigned, in progress, completed)
+    * The API review backlog and queue is publicly visible at https://github.com/orgs/kubernetes/projects/13
+
+3. Backlog
+    * An approver or moderator will adjust the prioritization of the issue in the backlog. Reviews are prioritized based on a number of factors:
+    * Whether the change is targeting the current milestone, a specific future milestone, or is untargeted
+    * The maturity level of the change (generally, GA > beta > alpha changes)
+    * Feedback from SIG leads / subproject leads on relative priorities
+    * Whether this is an initial review or re-review (or a review of a PR implementing an already-reviewed API in a KEP)
+    * Size/complexity of the change
+
+4. Assigned
+    * An approver or moderator will assign an approver (or potentially aspiring reviewer - see *training reviews* below for the aspiring reviewer workflow)
+    * Reviews are assigned based on reviewer capacity and domain knowledge
+    * Assignment of reviewers is done on the issue/PR itself using the normal `/assign` method (works seamlessly with existing github/PR dashboard queries)
+    * All API reviews assigned to an individual can be viewed in the project board ([example](https://github.com/orgs/kubernetes/projects/13/?card_filter_query=assignee%3Aliggitt)), for visibility on status, order, and reviewer load
+
+5. In Progress / Approved / Changes Requested / Rejected
+    * Reviews proceed like a normal KEP or PR review. Possible outcomes:
+    * Approval:
+      * Implementation PRs are tagged with `/lgtm /approve` and merged normally
+      * KEP PRs containing API designs can also be tagged with `/lgtm /approve`, but should explicitly note if API approval is being given. This approval should be linked to when later requesting review of the implementation PR, and should limit the scope of the implementation review to differences between the approved design and final implementation, problems encountered during implementation, and correctness of the implementation.
+      * The approved issue is archived in the review project board, and the `api-review` label is removed.
+    * Changes requested:
+      * Comments or questions are left on the PR or issue, and the reviewer notifies the submitter
+      * The reviewer moves the issue to "Changes Requested" in the review project board
+      * Once the requested changes are made, or questions are answered, the submitter notifies the reviewer, who moves the issue back to "In Progress"
+      * To the degree possible, complete sets of comments/changes should be requested and made, to avoid excessive back-and-forth cycles.
+    * Rejected:
```

Comment:
This should really never happen if it got through the KEP process, should it? We should be giving that feedback at that stage.

Reply:
yes, though this workflow applies equally to a review of an API design in a KEP itself. I agree that the "don't build this here/this way" outcomes would ideally be caught early.

the types of things I'd expect to get rejected at the PR stage are things whose goals were accepted as KEPs without concrete API designs, then whose implementations fundamentally violate an important API convention or principle